packages/wdio-browserstack-service/src/instrumentation/funnelInstrumentation.ts

```diff
@@ -151,3 +152,90 @@ function getReferrer(framework?: string) {
     const fullName = framework ? 'WebdriverIO-' + framework : 'WebdriverIO'
     return `${fullName}/${BSTACK_SERVICE_VERSION}`
 }
+
+const sendEvent = {
+    tcgDown: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgDownResponse', config),
+    invalidTcgAuth: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInvalidTcgAuthResponseWithUserImpact', config),
+    tcgAuthFailure: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgAuthFailure', config),
+    tcgtInitSuccessful: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgtInitSuccessful', config),
+    initFailed: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInitFailedResponse', config),
+}
+
+function handleUpgradeRequired(isSelfHealEnabled: boolean | undefined) {
+    if (isSelfHealEnabled) {
+        BStackLogger.warn('Please upgrade Browserstack Service to the latest version to use the self-healing feature.')
```

Comment:
Shouldn't we add which particular version to upgrade? That would help users easily grap the message

Reply:
We don't know which version of this service would be released in the future and when they'd be released. This is added in case AI team makes a major change in the healing flow and we deprecate the healing in the previous versions and support healing only on the latest version.